File: backend/src/main/java/com/music/education/Entity/ForumItem.java
package com.music.education.Entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ForumItem {
    private int id;
    private String title;
    private String content;
    private int author;
    private Timestamp time;
    private int reply_num;
    private int like_num;
    private String book_link;

    @Override
    public String toString() {
        return "ForumItem{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", content='" + content + '\'' +
                ", author=" + author +
                ", time=" + time +
                ", reply_num=" + reply_num +
                ", like_num=" + like_num +
                ", book_link=" + book_link +
                '}';
    }
}

File: app/src/main/java/com/example/musiceducation/entity/ForumItem.kt
package com.example.musiceducation.entity

import kotlinx.serialization.KSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Serializer
import kotlinx.serialization.encoding.Decoder
import kotlinx.serialization.encoding.Encoder
import java.sql.Timestamp
import java.text.SimpleDateFormat
import java.util.Locale


@Serializable
class ForumItem (
    val id: Int,
    val title: String,
    val content: String,
    val author_name: String,
    val author_avatar: String?,
    @SerialName("time") // Use SerialName to map to the correct JSON field name
    @Serializable(with = TimestampSerializer::class)
    @JsonPrimitive
    val time: Long,
    val reply_num: Int,
    val like_num: Int
)

annotation class JsonPrimitive

@Serializer(forClass = Long::class)
object TimestampSerializer : KSerializer<Long> {
    override fun deserialize(decoder: Decoder): Long {
        return decoder.decodeLong()
    }

    override fun serialize(encoder: Encoder, value: Long) {
        encoder.encodeLong(value)
    }
}